spark-operator/build.gradle

```diff
@@ -43,10 +43,17 @@ dependencies {
   implementation("io.dropwizard.metrics:metrics-core:$dropwizardMetricsVersion")
   implementation("io.dropwizard.metrics:metrics-jvm:$dropwizardMetricsVersion")
   compileOnly("org.apache.spark:spark-core_$scalaVersion:$sparkVersion") {
-    exclude group: 'com.squareup.okio'
-    exclude group: 'com.squareup.okhttp3'
+    exclude group: "com.github.luben"
+    exclude group: "io.netty", module: "netty-tcnative-boringssl-static"
+    exclude group: "io.netty", module: "netty-tcnative-classes"
     exclude group: "org.apache.logging.log4j"
+    exclude group: "org.fusesource.leveldbjni"
+    exclude group: "org.lz4"
+    exclude group: "org.rocksdb"
     exclude group: "org.slf4j"
+    exclude group: "org.xerial.snappy"
+    exclude group: 'com.squareup.okhttp3'
+    exclude group: 'com.squareup.okio'
```

Comment:
Why excluding these transitive dependencies? Aren't they used?

Reply:
`okio` is not excluded by this PR~ I sorted this part and it makes the lines confusing. Sorry about that.

Reply:
Technically, this PR removes mostly native libraries like the following, @viirya .
```
+    exclude group: "com.github.luben"
+    exclude group: "io.netty", module: "netty-tcnative-boringssl-static"
+    exclude group: "io.netty", module: "netty-tcnative-classes"
+    exclude group: "org.fusesource.leveldbjni"
+    exclude group: "org.lz4"
+    exclude group: "org.rocksdb"
+    exclude group: "org.xerial.snappy"
```

Reply:
I see. For example `org.rocksdb` should be used in structured streaming. So I think `spark-operator` doesn't need it.

Reply:
I got what you are confused here.

This is `Apache Spark Operator Docker Image` whose name is `spark-kubernetes-operator:0.1.0`. This only launches the user-provided image of `spark.kubernetes.container.image`. In case of `spark:4.0.0-preview1`, the user application is running on the isolated pod on top of `spark:4.0.0-preview`.

Reply:
In other words, this PR is not affecting any Spark application pods. It's only affecting `Spark Operator` pod.

Reply:
This is the running `Spark Operator` resources which are usually running on independent K8s namespaces, not inside the user's K8s namespace.
```
$ k get all
NAME                                             READY   STATUS    RESTARTS   AGE
pod/spark-kubernetes-operator-645d6dbd7d-lbbqx   1/1     Running   0          12m

NAME                 TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
service/kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   16m

NAME                                        READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/spark-kubernetes-operator   1/1     1            1           12m

NAME                                                   DESIRED   CURRENT   READY   AGE
replicaset.apps/spark-kubernetes-operator-645d6dbd7d   1         1         1       12m
```